src/SFA.DAS.EmployerAccounts.Web/Mappings/AgreementMappings.cs

```diff
@@ -35,6 +36,30 @@ public AgreementMappings()
 
             CreateMap<AccountDetailViewModel, AgreementInfoViewModel>()
                 .ConvertUsing(new AgreementInfoConverter());
+
+            CreateMap<EmployerAgreementDto, OrganisationAgreementViewModel>()
+                .ForMember(dest => dest.SignedDateText, opt => opt.Ignore())                             
+                .ForMember(dest => dest.AccountLegalEntityPublicHashedId, opts => opts.Ignore());
+
+            CreateMap<AgreementTemplateDto, AgreementTemplateViewModel>()
+                .ForMember(dest => dest.PublishedInfo, opt => opt.MapFrom(src => GetPublishedInfo(src)));                
+
+            CreateMap<AccountLegalEntityDto, AccountLegalEntityViewModel>();
+        }
+
+        private string GetPublishedInfo(AgreementTemplateDto agreementTemplateDto)
+        {
+            switch (agreementTemplateDto.VersionNumber)
+            {
+                case 1:
+                    return "Published 1 May 2017";                    
```

Comment:
Rather than returning a string (which is a UI concern) just set the PublishedDate  as a DateTime property of the AgreementTemplate.  Then the UI can add the extra text and format the date as per the UI requirements.

Reply:
done